internal/controller/dnsrecord_controller_test.go

```diff
@@ -527,6 +527,22 @@ var _ = Describe("DNSRecordReconciler", func() {
 		Expect(k8sClient.Create(ctx, dnsRecord2)).To(Succeed())
 
 		By("checking dnsrecord " + dnsRecord.Name + " and " + dnsRecord2.Name + " conflict")
+		Eventually(func(g Gomega) {
+			var oldRequeue, newRequeue time.Duration
+			var err error
+
+			oldRequeue, err = time.ParseDuration(dnsRecord.Status.ValidFor)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(dnsRecord), dnsRecord)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			newRequeue, err = time.ParseDuration(dnsRecord.Status.ValidFor)
+			g.Expect(err).NotTo(HaveOccurred())
+
+			g.Expect(oldRequeue.Milliseconds()).To(BeNumerically("<", newRequeue.Milliseconds()))
+		}, TestTimeoutMedium, time.Second).Should(Succeed())
+
```

Comment:
I'm not sure this test ("should detect a conflict and the resolution of a conflict") is providing any value, the two records that are being created don't conflict, tbh I'm not even sure off the top of my head how you can cause a conflict when using the inmem provider.

Point of this, as i understand it, is when two records are able to update the same field to a different value (aws default geo for example) but two A records will just merge these days and don't actually conflict. They will cause a write count increase, but that's expected

Reply:
Actually, you can get it to properly "conflict" if you set different TTL values on the two records. I would suggest making that change here.

Reply:
This test is, indeed, checks for nothing. Seems like it was relying on a bug or something with the inmemory provider to recreate a conflict. I'll change a test in here to do a conflict. 

As for the mess in the requeue process I've created #513 yesterday. It could be the excuse to revisit the logic and improve it :thinking:

Reply:
Changed the test to be a valid conflict test. With the changes, it didn't make sense to keep the `testEndpoints()` as it was, so I made a slight modification. It will also make it easier to adjust those in the future